src/lib/alloc.c

```diff
@@ -400,9 +443,14 @@ static void free_block(void *ptr)
 		block_map->first_free = block;
 
 #if DEBUG_BLOCK_FREE
-	/* memset the whole block incase some not aligned ptr*/
-	alloc_memset_region((void *)(block_map->base + block_map->block_size * block),
-			    block_map->block_size * (i - block), DEBUG_BLOCK_FREE_VALUE);
+	/* memset the whole block incase some not aligned ptr */
+	validate_memory(
+		(void *)(block_map->base + block_map->block_size * block),
+		block_map->block_size * (i - block));
```

Comment:
Yes and it is right result. We validate for possibility of freeing pointer second time. So if all memory has the pattern it means it was already freed (since it is zeroed on alloc).